### DISCUSSION TITLE
missing symbols: [error] * M16java.lang.ThreadD4joinuEO from scala cli

### DISCUSSION THREAD

**Discussion by doofin** [scala-native/scala-native#3535] [Q&A]
This errors show up when I tried  
> scala-cli --power package --native --docker main.sc

**Comment by WojciechMazur** [comment]
ScalaCLI is by default using the 0.4.x branch (version 0.4.14 or 15 right now) which does not yet support multithreading. However, you can use our 0.5.0-SNAPSHOT version which adds this capability. Simply add `--native-version=0.5.0-SNAPSHOT` however, be aware that external dependencies are not yet published for 0.5.x version and would fail to link

**Comment by doofin** [comment]
I tried the snapshot version, it builds successfully but complains: 

```
/app: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /app)
/app: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /app)

```
is this because of external dependencies?

**Comment by WojciechMazur** [comment]
I think this issue is due to version of C and C++ standard library used on your machine. What version of LLVM are you using (you can check using `clang --version`). 
I think it is only executed in  container, but build locally using your version of C/C++ library introducing a dynamic native dependency. You can run it without `--docker` flag to be sure. Probably the version of C in docker image is incompatible

**Comment by WojciechMazur** [comment]
I had exactly the issue when doing AWS lambdas with SN. https://youtu.be/r5GtrV3E68Q?t=3381 
Becouse the runtime of container in which binary was executed was different then the one in which the binary was build, it would introduce dependency to non existing versions of C/C++ standard library. I fixed it by building the binary in the container which had the same base image as the one used for it's execution and also by copying other external deps to layer attached to container running binary (https://github.com/WojciechMazur/ScalaNative-SAM-examples)
